packages/p2p/src/components/buy-sell/popup.jsx

```diff
@@ -0,0 +1,207 @@
+            send_currency,
+            receive_currency,
+            send_validation,
+            receive_validation,
+        };
+    }
+
+    render() {
+        const { ad, onCancel } = this.props;
+        const is_buy = ad.type === 'buy';
+        const {
+            initial_send,
+            initial_receive,
+            send_currency,
+            receive_currency
+        } = this.getInitialValues();
+
+        return (
+            <Fragment>
+                <div className='buy-sell__popup'>
+                    <div className='buy-sell__popup-header'>
+                        <div className="buy-sell__popup-header_wrapper">
+                            <h2 className='buy-sell__popup-header--title'>{`${ad.type} ${ad.asset}`}</h2>
+                            <IconClose className='buy-sell__popup-close_icon' onClick={onCancel} />
+                        </div>
+                    </div>
+                    <Formik
+                        validate={this.validatePopup}
+                        initialValues={{ send: initial_send, receive: initial_receive }}
+                        onSubmit={this.handleSubmit}
+                    >
+                        {({
+                            errors,
+                            isSubmitting,
+                            setFieldValue,
+                            handleChange,
+                        }) => (
+                            <Form noValidate>
+                                <ThemedScrollbars
+                                    autoHide
+                                    style={{ height: '307px' }}
+                                >
+                                    <div className='buy-sell__popup-content'>
+                                        <div className='buy-sell__popup-field_wrapper'>
+                                            <Field name='send'>
+                                                {({ field }) => (
+                                                    <Input
+                                                        {...field}
+                                                        data-lpignore='true'
+                                                        type='number'
+                                                        error={errors.send}
+                                                        label={localize('Send')}
+                                                        className='buy-sell__popup-field'
+                                                        placeholder={localize('Send amount')}
+                                                        trailing_icon={<span className='buy-sell__popup-field--trailing'>{send_currency}</span>}
+                                                        onChange={(e) => {
+                                                            const send = isNaN(e.target.value) ? 0 : e.target.value;
+                                                            const receive_amount = is_buy ?
```

Comment:
this should probably be moved to a function then `calculateReceiveAmount(is_buy)` or similar